ons_alpha/jinja2/component_overrides/footer/_macro.njk

```diff
@@ -0,0 +1,218 @@
+{% macro onsFooterNew(params) %}{# This is an override of ONS "components/footer/_macro.njk" #}
+    {% from "components/list/_macro.njk" import onsList %}
+    {% from "components/icon/_macro.njk" import onsIcon %}
+
+    {% if params.lang %}
+        {% set lang = params.lang %}
+    {% else %}
+        {% set lang = 'en' %}
+    {% endif %}
+
+    {% set onsLogo %}
+        {% if params.lang == 'cy' %}
+            {{-
+                onsIcon({
+                    "iconType": 'ons-logo-cy',
+                    "altText": 'Swyddfa Ystadegau Gwladol',
+                    "altTextId": 'ons-logo-cy-footer-alt'
+                })
+            -}}
+        {% else %}
+            {{-
+                onsIcon({
+                    "iconType": 'ons-logo-en',
+                    "altText": 'Office for National Statistics',
+                    "altTextId": 'ons-logo-en-footer-alt'
+                })
+            -}}
+        {% endif %}
+    {% endset %}
+
+    <footer class="ons-footer footer">
+        {% if params.footerWarning %}
+            {% from "components/panel/_macro.njk" import onsPanel %}
+            <div class="ons-footer__warning">
+                <div class="ons-container">
+                    {%
+                        call onsPanel({
+                            "variant": "warn",
+                            "classes": "ons-panel--warn--footer"
+                        })
+                    %}
+                        {{ params.footerWarning | safe }}
+                    {% endcall %}
+                </div>
+            </div>
+        {% endif %}
+
+        {% if params.button %}
+            {% from "components/button/_macro.njk" import onsButton %}
+            <div class="ons-footer__button-container ons-u-d-no@m">
+                {{
+                    onsButton({
+                        "id": params.button.id,
+                        "classes": params.button.classes if params.button.classes,
```

Comment:
Is `params.button.classes` meant to be a list? If so, this might be better:

```suggestion
                        "classes": params.button.classes or [],
```

Reply:
Again, same comment that it's their code that is duplicated here so I can add extra classes.